docs/ai/conceptual/understanding-tokens.md

```diff
@@ -0,0 +1,118 @@
+| Smaller tokens (character or subword tokenization) | - Allows the model to handle a wider range of inputs, such as unknown words, typos, or complex syntax<br>- Might allow the vocabulary size to be reduced, requiring less memory resources       | - A given text is broken into more tokens, requiring additional computational resources while processing<br>- Given a fixed token limit, the maximum size of the model's input and output is smaller |
+| Larger tokens (word tokenization)                  | - A given text is broken into fewer tokens, requiring less computational resources while processing<br>- Given the same token limit, the maximum size of the model's input and output is larger | - Might cause an increased vocabulary size, requiring more memory resources<br>- Can limit the models ability to handle unknown words, typos, or complex syntax                                      |
+
+## How LLMs use tokens
+
+After completing tokenization, LLMs assign an ID to each unique token.
+
+Consider the following text:
+
+> I heard a dog bark loudly at a cat
+
+After using a word tokenization method, a model could assign token IDs like this:
+
+- I (1)
+- heard (2)
+- a (3)
+- dog (4)
+- bark (5)
+- loudly (6)
+- at (7)
+- a (the "a" token is already assigned an ID of 3)
+- cat (8)
+
+By assigning IDs, text can be represented as a sequence of token IDs. The previous sentence would be represented as [1, 2, 3, 4, 5, 6, 7, 3, 8]. The sentence "I heard a cat" would be represented as [1, 2, 3, 8].
+
+As training continues, the model add any new tokens in the training text to it's vocabulary and assigns it an ID. For example:
+
+- meow (9)
+- run (10)
+
+The semantic relationships between the tokens can be analyzed using these token ID sequences. Multi-valued numeric vectors, known as [_embeddings_]<!-- (understanding-embeddings.md) -->, are used to represent these relationships. An embedding is assigned to each token based on how commonly it's used together with or in similar contexts to the other tokens.
+
+Once trained, a model can calculate an embedding for a text containing multiple tokens. The model tokenizes the text, then calculates an overall embeddings value based on the learned embeddings of the individual tokens. This technique can be used for semantic document searches or adding [memories](/semantic-kernel/memories/) to an AI.
+
+During output generation, the model predicts a vector value for the next token in the sequence. The model then selects the next token from it's vocabulary based on this vector value. In practice, the models calculates multiple vectors using various elements of the preceding tokens' embeddings. The model then evaluates all potential tokens from these vectors and selects the most probable one to continue the sequence.
```

Comment:
```suggestion
During output generation, the model predicts a vector value for the next token in the sequence. The model then selects the next token from it's vocabulary based on this vector value. In practice, the model calculates multiple vectors by using various elements of the previous tokens' embeddings. The model then evaluates all potential tokens from these vectors and selects the most probable one to continue the sequence.
```
After this change, this now means “the embeddings of the previous tokens,” which is what I assume the possessive punctuation means here. If we mean, “the last action of creating token embeddings,” we could use: “the preceding token embeddings.”